app/cafe.py

```diff
@@ -0,0 +1,28 @@
+import datetime
+
+from app.errors import (
+    NotVaccinatedError,
+    OutdatedVaccineError,
+    NotWearingMaskError
+)
+
+
+class Cafe:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def visit_cafe(self, visitor: dict) -> str:
+        current_date = datetime.date.today()
+
+        if "vaccine" in visitor:
+            expiration_date = visitor["vaccine"]["expiration_date"]
+            if expiration_date < current_date:
+                raise OutdatedVaccineError("Expiration date out of time")

```

Comment:
You could avoid using two nested if conditions in case you will use dict chaining with `get()` and using dict as a default return value, e.g.:
```suggestion
        if visitor.get("vaccine", {}).get("expiration_date", current_date) <= current_date:
            raise OutdatedVaccineError("Expiration date out of time")
```
You could modify the logic to satisfy the tests but keep the overall structure mentioned above.